lib/ngtcp2_acktr.c

```diff
@@ -293,6 +294,7 @@ void ngtcp2_acktr_recv_ack(ngtcp2_acktr *acktr, const ngtcp2_ack *fr) {
     min_ack = largest_ack - (int64_t)fr->ranges[i].len;
 
     for (;;) {
+      assert(ent != NULL);
```

Comment:
Isn't it suffice to just single assert above?
I could not reproduce this warning with gcc 14.2.0.  If this is gcc 14.2.1 specific bug, then lets wait for 14.2.2 which fixes the bug.

Reply:
`ent` is being assigned to in the loop `ent = ngtcp2_ringbuf_get(rb, j);` and gcc is unable to deduce that this function always returns non-null, hence the loop `assert`.

Reply:
> lets wait for 14.2.2 which fixes the bug

Ok, thank you for considering. Closing.